src/Data/Audience.elm

```diff
@@ -35,6 +64,17 @@ type alias Audience =
     }
 
 
+{-| Decoder of Audience record
+-}
+decoder : Decoder Audience
+decoder =
+    Decode.map4 Audience
+        (Decode.field "id" Decode.int)
+        (Decode.field "name" Decode.string)
+        (Decode.field "type" audienceTypeDecoder)
+        (Decode.field "folder" (Decode.nullable Decode.int))
```

Comment:
applicative style is usually simpler to extend though there is nothing wrong with using `map4`.

Reply:
Like everything, this is about tradeoffs and probably even certain experience biases. For Haskell folks when it comes to this type of composition Applicative style seems like the most elegant solution to the problem usually even though for instance @jhrcek also said on some occasions that he prefers mapN.

There is another issue with using mapN in bigger teams and that is that if 2 people in 2 PRs add different attribute it produces a bit more complicated conflict.

In our team, we use json-extra because at the time there was not decode-pipeline package yet and `andMap` is more general than functions from pipeline package which makes the assumption that applicative style is used over the object.

No super-strong preferences from my side though. I think this is not that crucial thing usually. One probably favors the consistency over anything else when it comes to things like this.

The only thing this was raised several times was probably that [hints in the requirements mentions this](https://github.com/GlobalWebIndex/challenge-elm/blob/master/FIRST_STEP.md#hints)